arch/arm/core/aarch32/cortex_m/mpu/arm_mpu.c

```diff
@@ -34,9 +49,9 @@ static inline u8_t get_num_regions(void)
 	 * have a fixed number of 8 MPU regions.
 	 */
 	return 8;
```

Comment:
```
#if defined(CONFIG_CPU_CORTEX_M0PLUS) || \
	defined(CONFIG_CPU_CORTEX_M3) || \
	defined(CONFIG_CPU_CORTEX_M4)
	/* Cortex-M0+, Cortex-M3, and Cortex-M4 MCUs may
	 * have a fixed number of 8 MPU regions.
	 */
	return 8;
```

While we are at it, I wonder if we should get rid of this hard coded number and use the device tree value.

For now, we are omitting the device tree MPU node for the MPU-equipped Cortex-M0(+)/-M3/-M4 devices and this can give false impression that the devices do not support MPU.

As for the region number validation, maybe we can add a `BUILD_ASSERT` for it.

cc @ioannisg

Reply:
This hard coded value must exist until all Cortex-M SoCs in the tree have a DTS node for MPU - right now this is the case for Cortex-M7 -M23, and -M33 SoCs which may have either 4, 8, or 16 MPU regions. Until we fully transition to DTS-based representation of MPU, this hard-coded return is required.

But even then, I see a benefit of this for out-of-tree SoC M0+, M3, or M4 definitions that won't need to bother to add a DTS node for MPU.

> For now, we are omitting the device tree MPU node for the MPU-equipped Cortex-M0(+)/-M3/-M4 devices and this can give false impression that the devices do not support MPU.

No objections with adding MPU nodes on all Cortex-M SoCs. But I don't think we hint that there's no MPU support; after all, this comment

```
/* Cortex-M0+, Cortex-M3, and Cortex-M4 MCUs may
 * have a fixed number of 8 MPU regions.
 */
```
clarifies the things:

Reply:
Yes, if we are to remove that hard-coded check, we need to add device tree MPU node to all MPU-equipped SoCs.

The main concern with the completely omitting the MPU device tree node for the fixed-region-count MPU-equipped SoCs is that, when looking at only dts, we have no way of differentiating MPU-less SoC vs. MPU-equipped SoC.

After all, the purpose of device tree is to describe hardware, and omitting the MPU node because we do not need params/configs from it is less than ideal.

Reply:
I agree with should add the nodes to the DTS for v6m SoCs that support it and phase out the hardcoded value.  However, I think this is outside the scope of this PR :).